currency/storage.go

```diff
@@ -116,15 +116,20 @@ func (s *Storage) RunUpdater(overrides BotOverrides, settings *MainConfiguration
 		log.Debugf("Setting up currency analysis system with Coinmarketcap...")
 		c := &coinmarketcap.Coinmarketcap{}
 		c.SetDefaults()
-		c.Setup(coinmarketcap.Settings{
+		err := c.Setup(coinmarketcap.Settings{
 			Name:        settings.CryptocurrencyProvider.Name,
 			Enabled:     true,
 			AccountPlan: settings.CryptocurrencyProvider.AccountPlan,
 			APIkey:      settings.CryptocurrencyProvider.APIkey,
 			Verbose:     settings.CryptocurrencyProvider.Verbose,
 		})
-
-		s.currencyAnalysis = c
+		if err != nil {
+			log.Errorf("Unable to setup CoinMarketCap analysis. Error: %s", err)
```

Comment:
Golang loves lowercase starting error messages when it isn't a acronym or pronoun

Reply:
https://github.com/golang/go/wiki/Errors; for ref, it does say that, but seeing as this is used as a prefix for the err, all that is needed is the underlying exported error type should be lower case?

Reply:
To add a little more

The reason behind this comes from https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Normally you would pass the error down the chain until its finally displayed

so you may end up with something 
```go
return fmt.Errorf("Something bad")
```

then later down the track
```go 
log.Errorf("Some error: %v", err)
```

That will cause it to display 
```
Some error: Something bad
```

if you are just logging the output it's fine but Goland code detection does not pick up on this and just warns either way (same with some other linters)

Reply:
This is just the prefix of displaying an error. All throughout our code base we use capital letters for the initial prefix of things and errors being lower case as per go spec, that's why our linters don't mind.

Reply:
An issue for the engine branch then 🦀 😄